### ISSUE TITLE
Hanging on startup (v11.2403.6.0)

### ISSUE THREAD

**Issue by SanderBouwhuis** [microsoft/calculator#2173]
**Describe the bug**
When I start the calculator, it sometimes hangs. It shows a blue window with the calculator icon, but never continues to the calculator. I think it started happening after this month's Windows Tuesday updates.
It seems to happen more just after I restart Windows.

**Steps To Reproduce**
1. Start the calculator
2. The calculator hangs

**Expected behavior**
It should continue to start

**Screenshots**
![2024-05-19 12-09-01 670x500 Program_Manager](https://github.com/microsoft/calculator/assets/10495173/4d7c7c7a-fc0e-41dc-89bf-274e18a7fdd7)
![2024-05-19 12-26-04 1154x513 CalculatorApp exe_(23036)_Properties](https://github.com/microsoft/calculator/assets/10495173/e2d23b30-2735-4a45-8678-2eb3881525c1)

**Dump file**
https://mega.nz/file/no1kCRQR#qCbgUNPJndry6f1lDrYxIBQoB5Jt86mO7OhPkQTLUcA

**Device and Application Information**
 - OS Build: Windows 10
 - Architecture: 64-bit
 - Application Version: v11.2403.6.0
 - Region: ?!? Europe ?!?
 - Dev Version Installed: ?!? Don't know what this means ?!?

Run the following commands in Powershell and copy/paste the output.
" - OS Build: $([Environment]::OSVersion.Version)"
Major  Minor  Build  Revision
10     0      19045  0
" - Architecture: $((Get-AppxPackage -Name Microsoft.WindowsCalculator).Architecture)"
X64
" - Application Version: $((Get-AppxPackage -Name Microsoft.WindowsCalculator).Version)"
11.2403.6.0
" - Region: $((Get-Culture).Name)"
en-US
" - Dev Version Installed: $($null -ne (Get-AppxPackage -Name Microsoft.WindowsCalculator.Dev))"
False

**Comment by greenythebeast** [comment]
Also having this issue. Any help would be appreciated!

**Comment by hanzhang54** [comment]
Thanks for reporting this issue, and we are looking into this. (Internal bug id # 51587569)

**Comment by johnrs** [comment]
Same problem.  Started about a week ago.

I found that if I close it (click on the "x"), then start it again there is about a 5 sec delay, then it appears and works.

**Comment by greenythebeast** [comment]
> Same problem. Started a week or two ago. I'm running Windows 10.
> 
> I found that if I close it (click on the "x"), then start it again there is about a 5 sec delay, then it appears and works.

Yup, closing the app and waiting a few seconds to start it up again usually works.

**Comment by johnrs** [comment]
Still no fix?

**Comment by SanderBouwhuis** [comment]
It is totally ridiculous that an application as simple as a calculator worked better in Windows v3.11 than in Windows 10!

Is there a way to download an older version?

**Comment by Prow7** [comment]
I have the same issue. It started happening earlier this year, probably in May.

The issue happens almost 100% of the time when using the calculator button on my keyboard when launching calculator for the first time after booting.

It's much less likely to happen when I launch calculator from the start menu.

In any case, the fix is what others have posted; close calculator and launch it again. What's annoying is it takes about 5 seconds before I can launch calculator again after closing it.

**Comment by SanderBouwhuis** [comment]
It happens almost every time for me. Now we can see what the difference is between professional commercial software and amateur open source projects. I guess you get what you pay for.

**Comment by CVIEWarch** [comment]
I have the same problem since two months. It never starts correctly. I have to close it then press a couple times the calc. icon and after a delay it starts. This is so ridiculous

**Comment by myyamma** [comment]
Having the exact same issue, started several months ago. 

Obviously still no resolution?

**Comment by CVIEWarch** [comment]
Has anyone find an alternative app that works? I searched for other calculator apps but there is none similar to the windows calculator. 
This is a huge problem for windows and nobody seems to care fixing it! The moment I need to make an operation, it takes more than half a minute to open the calculator.. I have to open it, get a blue screen (as described above) , then I have to close it and then hit the calculator keyboard button 4 or 5 times and after a while the application starts.

**Comment by SanderBouwhuis** [comment]
This is the last thing I'll try. This calculator project is dead and ruined.
@CVIEWarch, how do I even uninstall this application? It's not in Programs and Features. I read online I had to go to the 'Windows Store' application and look for it in my library. But, how do I uninstall it?!?

**Comment by hanzhang54** [comment]
Hi all, we are still unable to repro this and collect enough information for investigation. If someone is still experiencing the issue, please file the issue in Feedback Hub app, which would be helpful for information collection. Thanks!

**Comment by johnrs** [comment]
Uninstall then reinstall (without any rebooting) seems to have worked for me.

**Comment by CVIEWarch** [comment]
5 days later and it still works fine after following these steps. Maybe the last windows updates fixed the issue
![image](https://github.com/user-attachments/assets/7e70e8eb-dadc-4319-9e1c-db7ce18462a6)


> I tried the following today and it seems it has fixed the issue:
> 
> 1. I uninstalled the calculator.
> 2. I restarted the pc.
> 3. I run the Disk Cleanup in the drive that windows are installed
> 4. I restarted again and then I downloaded and installed the windows calculator.
> 5. Select Start Windows logo Start button > Settings Gear-shaped Settings icon > Apps > Apps & features.
>    6 Select the Calculator app.
>    7 Select the Advanced options link under the name of the app. On the page that opens, select **Repair** .

**Comment by omgitsraven** [comment]
Having the same situation here:
- launching from Start menu seems fine
- launching from keyboard calculator button gives the blue icon screen if I haven't launched it recently

Interestingly:  writing an AutoHotKey script that intercepts a different keystroke and then calls "run calc.exe", which I was hoping would work around this, in fact gives the exact same blue icon screen behaviour!

**Comment by brinkdinges** [comment]
> Having the same situation here:
> 
> * launching from Start menu seems fine
> * launching from keyboard calculator button gives the blue icon screen if I haven't launched it recently
> 
> Interestingly: writing an AutoHotKey script that intercepts a different keystroke and then calls "run calc.exe", which I was hoping would work around this, in fact gives the exact same blue icon screen behaviour!

Interesting discovery, it behaves the same here. It starts correctly from the Start menu, I close it, then press the calculator button on my Microsoft Ergonomic Keyboard 4000 and the blue screen calculator appears. Since I am running MS Mouse and Keyboard center, I remapped the calculator button to launching an app, in this case the calculator. And that one works 🤨

**Comment by omgitsraven** [comment]
I'm not sure if this is useful information, but:  somehow this problem has suddenly gone away at some point since writing my previous comment, despite it being the case for *years* before that point.     I noticed it was finally working again a few weeks ago, and it hasn't happened since.    Will report back if the problem re-emerges...

**Comment by websnail** [comment]
Well, blow me... same here... I have a Microsoft Ergonomic keyboard and remapping the keyboard button to the app, in MS Mouse and Keyboard center, rather than just leaving it as default and it works.

**Comment by johnrs** [comment]
How did you do it?  I don't see the calculator key as available for remapping.  Also, I searched and couldn't find a VK number for it.

I did see how to remap the "1", "2", "3" user keys at the very top, left.  So I remapped one of those to the calculator.  It does start it, but I'll have to wait a while to see if it starts correctly.

**Comment by johnrs** [comment]
The 14.41 version of the Microsoft Mouse and Keyboard Center didn't fix the problem.  After uninstalling the old version, I installed the new version.  The keys I had remapped to Calculator were still remapped.  So the uninstall doesn't do away with the mappings.

The new version had the same problem with the Calculator now wanting to start till you tried again, about 5 seconds or so later.  BUT it also introduced another problem.  It reset my external monitors about every other minute, and sometimes it didn't recover automatically, thus requiring a manual reboot of the monitors.  It kept doing this every 2 minutes or so for about 15 minutes.  So I uninstalled it.

Monitors were back to normal, and guess what?  The keys were still mapped to the Calculator.  So the Center program must change a table somewhere in Windows but it isn't needed after that.  Then I found out that the Calculator works just fine now, from any of the 3 keys I mapped to it, including the original Calculator key itself.

So all along all I had to do was to get rid of the buggy Microsoft Center program.

Sounds like this is part of a long-term plan to get people to stop using keyboards and use AI instead.  <grin>
